R/Reporter.R

```diff
@@ -159,16 +161,165 @@ Reporter <- R6::R6Class( # nolint: object_name_linter.
+              block$to_list()
+            },
+            TableBlock = {
+              file.copy(block$get_content(), output_dir)
+              block$to_list()
+            },
+            NewpageBlock = list(),
+            NULL
+          )
+          new_block <- list()
+          new_block[[block_class]] <- cblock
+          new_blocks <- c(new_blocks, new_block)
+        }
+        new_card <- list()
+        new_card[["blocks"]] <- new_blocks
+        new_card[["metadata"]] <- card$get_metadata()
+
+        u_card <- list()
+        u_card[[card_class]] <- new_card
+        json$cards <- c(json$cards, u_card)
+      }
+
+      cat(jsonlite::toJSON(json, auto_unbox = TRUE, force = TRUE),
+        file = file.path(output_dir, "Report.json")
+      )
+      output_dir
+    },
+    #' @description Create/Recreate a Reporter from a dir with JSON file and static files
+    #' @param output_dir `character(1)` a path to the directory
+    #' @return invisibly self
+    #' @examples
+    #' reporter <- Reporter$new()
+    #' tmp_dir <- file.path(tempdir(), "jsondir")
+    #' dir.create(tmp_dir)
+    #' unlink(list.files(tmp_dir, recursive = TRUE))
+    #' reporter$to_jsondir(tmp_dir)
+    #' reporter$from_jsondir(tmp_dir)
+    from_jsondir = function(output_dir) {
+      checkmate::assert_directory_exists(output_dir)
+      checkmate::assert_true(length(list.files(output_dir)) > 0)
+      dir_files <- list.files(output_dir)
+      which_json <- grep("json$", dir_files)
+      json <- jsonlite::read_json(file.path(output_dir, dir_files[which_json]))
+      if (json$version == "1") {
+        new_cards <- list()
+        cards_names <- names(json$cards)
+        cards_names <- gsub("[.][0-9]*$", "", cards_names)
+        for (iter_c in seq_along(json$cards)) {
+          card_class <- cards_names[iter_c]
+          new_card <- switch(card_class,
+            ReportCard = ReportCard$new(),
+            TealReportCard = TealReportCard$new()
+          )
+          blocks <- json$cards[[iter_c]]$blocks
+          metadata <- json$cards[[iter_c]]$metadata
+          blocks_names <- names(blocks)
+          blocks_names <- gsub("[.][0-9]*$", "", blocks_names)
+          for (iter_b in seq_along(blocks)) {
+            block_class <- blocks_names[iter_b]
+            block <- blocks[[iter_b]]
+            cblock <- switch(block_class,
+              TextBlock = TextBlock$new()$from_list(block),
+              PictureBlock = {
+                new_file_path <- tempfile(fileext = ".png")
+                file.copy(file.path(output_dir, basename(block$path)), new_file_path)
+                PictureBlock$new()$from_list(list(path = new_file_path))
+              },
+              TableBlock = {
+                new_file_path <- tempfile(fileext = ".RDS")
+                file.copy(file.path(output_dir, basename(block$path)), new_file_path)
+                TableBlock$new()$from_list(list(path = new_file_path))
```

Comment:
Very important part. 
When the FileBlock is finalized then its content path is removed so we have to be careful here.
We have to use an external files for the PictureBlock and TableBlock content paths. If we add here the archive paths then we will break the process, as then the archive files will be removed when the blocks will be removed.